ui/src/SensitiveData.js

```diff
@@ -0,0 +1,161 @@
+import React, { Component } from "react";
+import "@fontsource/roboto";
+import {
+  Button,
+  Dialog,
+  DialogContent,
+  FormControl,
+  InputLabel,
+  MenuItem,
+  Select,
+  TextField,
+} from "@material-ui/core";
+
+
+class SensitiveData extends Component {
+  state = {
+    columnToUpdate: "",
+    newValue: "",
+    newValueValidationError: "",
+    showDialog: false,
+    allowableSensitiveDataColumns: []
+  };
+
+  openDialog = () => {
+    this.refreshDataFromBackend()
+    this.setState({
+      showDialog: true,
+    });
+  };
+
+  closeDialog = () => {
+    this.setState({
+      columnToUpdate: "",
+      newValue: "",
+      newValueValidationError: "",
+      showDialog: false,
+    });
+  };
+
+  getSensitiveSampleColumns = async () => {
+    const response = await fetch(`/api/surveys/${this.props.surveyId}`);
+    if (!response.ok) {
+      return;
+    }
+
+    const surveyJson = await response.json();
+    const sensitiveColumns = surveyJson.sampleValidationRules
+      .filter((rule) => rule.sensitive)
+      .map((rule) => rule.columnName);
+
+    return sensitiveColumns;
+  };
+
+  onSensitiveDataColumnChange = (event) => {
+    this.setState({ columnToUpdate: event.target.value });
+  };
+
+  onChangeValue = (event) => {
+    this.setState({
+      newValue: event.target.value,
+    });
+  };
+
+  refreshDataFromBackend = async () => {
```

Comment:
as in earlier comment, I would move this into the `openDialog` function so the dialog doesn't even display until it has the data it needs... but that's just a style thing